src/models/semester.ts

```diff
@@ -3,28 +3,31 @@ import { Document } from 'dynamoose/dist/Document';
 import Course, { CourseInterface } from './course';
 
 export const semesterSchema = new dynamoose.Schema(
-    {
-        id: String,
-        name: String,
-        courses: {
-            type: Set,
-            schema: [Course],
-        },
-        owner: {
-            type: String,
-            required: true,
-        },
-    },
-    {
-        timestamps: true,
-    }
+	{
+		id: {
+			type: String,
+			hashKey: true,
+		},
+		name: String,
+		courses: {
+			type: Set,
+			schema: [Course],
+		},
+		owner: {
+			type: String,
+			required: true,
+		},
+	},
+	{
+		timestamps: true,
+	}
 );
 
 export interface SemesterInterface extends Document {
-    id: string;
-    name: string;
-    courses: CourseInterface[];
-    owner: string;
+	id: string;
+	name: string;
+	courses?: Set<CourseInterface>;
```

Comment:
I don't want to make this optional, I think we should use an empty set instead when there is no courses. It will simplify our code because we won't have to check if there are courses, and the same code that handles a list of courses will likely handle an empty list fine. This is pretty much the null object principle  https://sourcemaking.com/design_patterns/null_object

Reply:
I did this because DynamoDB doesn't support empty sets. A different workaround would be to pass a set containing an empty string when creating a semester as this is supported: `"courses": [""]`

Reply:
I am assuming then that this does not apply https://aws.amazon.com/about-aws/whats-new/2020/05/amazon-dynamodb-now-supports-empty-values-for-non-key-string-and-binary-attributes-in-dynamodb-tables/

I am unsure what we should do for this, but maybe before we send the response back to the client, we undefined to an empty array for easier consumption

Reply:
Exactly 😄